File: serverSpring/src/main/java/com/viecz/server/model/Category.java
package com.viecz.server.model;

import jakarta.persistence.*;
import lombok.*;

/**
 * Category entity representing job categories in the Viecz platform.
 * Categories help organize and filter jobs.
 */
@Entity
@Table(name = "categories")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(name = "name_vi", nullable = false, length = 50)
    private String nameVi;

    @Column(length = 50)
    private String icon;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;

    // Relationships
    // @OneToMany(mappedBy = "category")
    // private List<Job> jobs;
}

File: android/app/src/main/java/com/viecz/vieczandroid/data/models/Category.kt
package com.viecz.vieczandroid.data.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Category(
    @Json(name = "id") val id: Int,
    @Json(name = "name") val name: String,
    @Json(name = "name_vi") val nameVi: String,
    @Json(name = "icon") val icon: String? = null,
    @Json(name = "is_active") val isActive: Boolean = true
)

@JsonClass(generateAdapter = true)
data class CategoriesResponse(
    @Json(name = "categories") val categories: List<Category>
)
